### ISSUE TITLE
Error using deepcopy

### ISSUE THREAD

**Issue by jpbascur** [vtraag/leidenalg#5]
Hi, I run the following code
```
import copy
import leidenalg as la
import pickle
G = pickle.load(open('G.pickle', "rb" ))
partition = la.find_partition(G, la.CPMVertexPartition,resolution_parameter = 0.0004)
partition_copy  = copy.deepcopy(partition)
```
And I get this error
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-68-73b2513dd65d> in <module>()
      1 partition  = la.find_partition(G, la.CPMVertexPartition, resolution_parameter = 0.0004)
----> 2 partition_copy = copy.deepcopy(partition)

~\AppData\Local\Continuum\anaconda3\lib\copy.py in deepcopy(x, memo, _nil)
    178                     y = x
    179                 else:
--> 180                     y = _reconstruct(x, memo, *rv)
    181 
    182     # If is its own copy, don't memoize.

~\AppData\Local\Continuum\anaconda3\lib\copy.py in _reconstruct(x, memo, func, args, state, listiter, dictiter, deepcopy)
    278     if state is not None:
    279         if deep:
--> 280             state = deepcopy(state, memo)
    281         if hasattr(y, '__setstate__'):
    282             y.__setstate__(state)

~\AppData\Local\Continuum\anaconda3\lib\copy.py in deepcopy(x, memo, _nil)
    148     copier = _deepcopy_dispatch.get(cls)
    149     if copier:
--> 150         y = copier(x, memo)
    151     else:
    152         try:

~\AppData\Local\Continuum\anaconda3\lib\copy.py in _deepcopy_dict(x, memo, deepcopy)
    238     memo[id(x)] = y
    239     for key, value in x.items():
--> 240         y[deepcopy(key, memo)] = deepcopy(value, memo)
    241     return y
    242 d[dict] = _deepcopy_dict

~\AppData\Local\Continuum\anaconda3\lib\copy.py in deepcopy(x, memo, _nil)
    167                     reductor = getattr(x, "__reduce_ex__", None)
    168                     if reductor:
--> 169                         rv = reductor(4)
    170                     else:
    171                         reductor = getattr(x, "__reduce__", None)

TypeError: can't pickle PyCapsule objects
```

**Comment by vtraag** [comment]
Thanks for the report. Indeed it is not possible to pickle the partition. I'll have a look at how to implement the [`__copy__` and `__deepcopy__`](https://docs.python.org/2/library/copy.html) functionality properly.

**Comment by jpbascur** [comment]
You are welcome

**Comment by vtraag** [comment]
Now implemented in 1b77e7d15b87721e5ac53b5c60f00a36c99bdc9a, will probably be included in next release.
